### ISSUE TITLE
Usages analysis incorrect when referring var with dot

### ISSUE THREAD

**Issue by mk** [clj-kondo/clj-kondo#1919]
Given the following namespace:

```clojure
(ns headless-ui
  (:require ["@headlessui/react" :refer [Dialog Dialog.Overlay]]))

(defn modal-overlay [_]
  (into [:> Dialog.Overlay {:class-name "fixed"}]))
```

The following warning is emitted:
```sh
$ clj-kondo --lint src/ductile/ui/headless_ui.cljs 
src/ductile/ui/headless_ui.cljs:2:49: warning: #'@headlessui/react/Dialog.Overlay is referred but never used
linting took 11ms, errors: 0, warnings: 1
```

This is incorrect as `Dialog.Overlay` overlay is used, but `Dialog` is not.

**Comment by borkdude** [comment]
With nbb this works, but with shadow-cljs I get:

```
cljs.core.prn.cljs$core$IFn$_invoke$arity$variadic(cljs.core.prim_seq.cljs$core$IFn$_invoke$arity$2([Foo.Bar], 0));
                                                                                                     ^

ReferenceError: Foo is not defined
```

so there it indeed seems to be necessary to import `Foo.Bar` It's not obvious to me why though since this just works fine in JS:

``` javascript
import { Foo } from './foo.js';

console.log(Foo);
console.log(Foo.Bar);
```

and this doesn't even run:

``` javascript
import { Foo, Foo.Bar } from './foo.js';

console.log(Foo);
console.log(Foo.Bar);
```

I feel that this is a short-coming of CLJS and I'm not sure if doing `:refer [Foo, Foo.Bar]` happens to work by accident or is intended to be supported. I'll ask in cljs-dev.

**Comment by borkdude** [comment]
Made a shadow-cljs issue:

https://github.com/thheller/shadow-cljs/issues/1076

**Comment by mk** [comment]
@borkdude thank you!

> So I think when you remove Dialog.Overlay from the imports, this will still work?

This broke the build which is how I discovered the issue in the first place.
